src/System Application/App/AI/src/Azure OpenAI/AzureOpenAIImpl.Codeunit.al

```diff
@@ -473,15 +488,24 @@ codeunit 7772 "Azure OpenAI Impl"
             Arguments.ReadFrom(Token.AsValue().AsText());
 
         if ChatMessages.GetFunctionTool(FunctionName, AOAIFunction) then
-            if TryExecuteFunction(AOAIFunction, Arguments, FunctionResult) then begin
+            if ChatMessages.GetToolInvokePreference() in [Enum::"AOAI Tool Invoke Preference"::InvokeToolsOnly, Enum::"AOAI Tool Invoke Preference"::Automatic] then
+                if TryExecuteFunction(AOAIFunction, Arguments, FunctionResult) then begin
+                    AOAIFunctionResponse.SetFunctionCallingResponse(true, true, AOAIFunction.GetName(), FunctionId, FunctionResult, '', '');
+                    AOAIOperationResponse.AddFunctionResponse(AOAIFunctionResponse);
+                    exit(true);
+                end else begin
+                    AOAIFunctionResponse.SetFunctionCallingResponse(true, false, AOAIFunction.GetName(), FunctionId, FunctionResult, GetLastErrorText(), GetLastErrorCallStack());
+                    AOAIOperationResponse.AddFunctionResponse(AOAIFunctionResponse);
+                    exit(true);
+                end
+            else begin
                 AOAIFunctionResponse.SetFunctionCallingResponse(true, true, AOAIFunction.GetName(), FunctionId, FunctionResult, '', '');
```

Comment:
I haven't gone through the whole PR, but my thought here would be.
`AOAIOperationResponse.IsSuccess` = `true`
`AOAIOperationResponse.IsFunctionCall` = `true`
`AOAIFunctionResponse.IsInvoked` = `false`
`AOAIFunctionResponse.IsSuccess` = `false`
We should have a distinction between a successful function call and if we invoked the function call. If we try and mix the two distinctions into one, it would be misleading for the majority of developers.